src/main/java/picard/analysis/CollectWgsMetricsWithNonZeroCoverage.java

```diff
@@ -186,7 +187,7 @@ protected WgsMetrics generateWgsMetrics(final IntervalList intervals,
 
     @Override
     protected WgsMetricsCollector getCollector(final int coverageCap, final IntervalList intervals) {
-        assert(coverageCap == this.collector.coverageCap);
+        ValidationUtils.validateArg(coverageCap == this.collector.coverageCap,()->"coverageCap has to be the same as the internal coverageCap, found " + coverageCap + " and " + this.collector.coverageCap);
```

Comment:
Please reformat the changed lines in the IDE. In this case I would expect to see spaces around the `->` operator, and the line length seems long to me.